Range collection (not full)

olehkhalus opened the pull request.

olehkhalus committed:
First commit

olehkhalus committed:
Range collection

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,117 @@
+    private HashSet<T> range;
+    private static final Range<Byte> rByte = new Range();
+    private static final Range<Short> rShort = new Range();
+    private static final Range<Integer> rInt = new Range();
```
You have to be able to add arbitrary number of types, you don't have to create all of them here.
Let's say I'm going to create a new numeric type that has only odd number, you should be able to handle it without changing your code.

olehkhalus committed:
Update Range collection

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,124 @@
 
     public boolean isEmpty() {
-        return false;
+        if(size()==1){
```
What is that?

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,124 @@
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
```
Don't user wildcard import (you can change it in Intellij configuration)

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,124 @@
+
+    public static Range<Short> of( short element1, short element2){
+        Range <Short> rShort = new Range<Short>();
+        for(short i=element1;i<=element2;i++){
```
Think how you can improve code by calling of(T start, T end, Function<T, T> function)

olehkhalus committed:
Remove wildcard`s import
Delete all basic Range<> methods
update Range<Integer> and Range<Float> methods

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,103 @@
+        return rInt;
+    }
+
+    public static Range<Float> of( float element1, float element2){
```
Think about how to call method **Range.of(T start, T end, Function<T, T> function)** instead of writing code here.

olehkhalus committed:
Update Range collection

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,119 @@
+            return rT;
+        }
+
+        if(start instanceof Integer && end instanceof Integer){
```
Don't do that, create for Integer and Float separate methods as it was.
But, instead of implementing all the logic inside those methods (for Float and Integer) just call method of(T start, T end, Function<T, T> function)

george-stotle replied:
let me give you an idea, inside of method of(Integer, Integer) call Range.of(start, end, new Function<Integer, Integer>() {

    @Override
    public Integer apply(Integer o) {
        return o+1;
    }
});

olehkhalus committed:
Update Range collection

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,102 @@
+        Range<Float> rFloat = Range.of(start, end, new Function<Float, Float>() {
+            @Override
+            public Float apply(Float aFloat) {
+                return (aFloat+0.1f);
```
Remove redundant brackets

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,102 @@
+                    return rInt;
+        }
+
+         static Range<Float> of(float start, float end){
```
Format code, also I think there should be another access modifier

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,102 @@
+                    return (integer +1);
+                }
+            });
+                    return rInt;
```
Format code.

george-stotle commented on src/main/java/co/inventorsoft/academy/collections/model/Range.java:
```diff
@@ -1,60 +1,102 @@
+    static <T extends Comparable> Range<T> of(T start, T end, Function<T, T> function) {
+        Range<T> rT = new Range<>();
+        if (start.equals(end)) {
+            return rT;
```
Have only one return statement per method.

olehkhalus committed:
Update Range collection
